aya/src/bpf.rs

```diff
@@ -58,7 +60,243 @@ pub(crate) struct bpf_map_def {
     pub(crate) map_flags: u32,
     // optional features
     pub(crate) id: u32,
-    pub(crate) pinning: u32,
+    pub(crate) pinning: BpfPinningType,
+}
+
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub(crate) enum BpfPinningType {
+    PinNone = 0,
+    PinByName = 1,
+}
+
+impl TryFrom<u32> for BpfPinningType {
+    type Error = BpfError;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(BpfPinningType::PinNone),
+            1 => Ok(BpfPinningType::PinByName),
+            n => Err(BpfError::UnexpectedPinningType { name: n }),
+        }
+    }
+}
+
+impl Default for BpfPinningType {
```

Comment:
Samme for this I don't think it's used

Reply:
This is needed because we have `#[derive(Default)]` on `bpf_map_def`. As `BpfPinningType` is a member, we need an `impl Default` as we can't derive one.

Reply:
Oh yep you're right